### PR TITLE
Improved ExplainTests.test_basic().

### PR DESCRIPTION
**Pull request by felixxm**
`QuerySet.select_for_update()` is not supported by all databases. Moreover it cannot be used outside of a transaction, e.g. on MySQL with MyISAM storage engine it crashes:
```
select_for_update cannot be used outside of a transaction.
```

### CODE REVIEW

File: tests/queries/test_explain.py [hunk 1]
```diff
@@ -31,13 +32,19 @@ def test_basic(self):
...
             for format in all_formats:
                 with self.subTest(format=format, queryset=idx):
-                    with self.assertNumQueries(1), CaptureQueriesContext(
-                        connection
-                    ) as captured_queries:
-                        result = queryset.explain(format=format)
+                    with CaptureQueriesContext(connection) as captured_queries:
+                        if queryset.query.select_for_update:
+                            with transaction.atomic():
+                                result = queryset.explain(format=format)
```
[the remark this answers is not in the document]
Comment (felixxm) [reply tests/queries/test_explain.py:38]: It depends on the database, in most of cases you will have an extra queries related with creating savepoints. That's why  I switched to `any()` and removed checking the number of queries when we start a new transaction.

[the remark this answers is not in the document]
Comment (hramezani) [reply tests/queries/test_explain.py:38]: OK, Thanks for explanations

**Review by hramezani — APPROVED** [review]
👍

**Comment by felixxm** [comment]
@hramezani Thanks :+1:

**Pull request merged**